fails:Kernel.fail is a private method
